### DISCUSSION TITLE
Qute: How to test type

### DISCUSSION THREAD

**Discussion by trixpan** [quarkusio/quarkus#44826] [Q&A] [answered]
After reading the guide I couldn't find the answer.

What is the simplest way to iterate over something like:

```
   Map<String, Object> attributes = new java.util.HashMap<>(Map.of());
   ...
   instance = instance.data("attributes", attributes);
```

Where some attributes can be `String` and others `List` or `Map`?

Using something as

```
          {#for attribute in attributes}
            <tr>
                <td>{attribute.key}</td>
                {#if attribute.value.size}
                    "ohh"
                {/if}
            </tr>
            {/for}
```

results in

`Property "size" not found on the base object "java.lang.String" in expression {attribute.value.size}`

[a comment by an automated account is not shown]

**Comment by mkouba** [comment, marked as answer]
You can use default values and safe expressions in order to suppress the error, see also [Strict Rendering](https://quarkus.io/guides/qute-reference#strict_rendering). In your case, `{#if attribute.value.size??}` should do the trick. It's a shorthand notation for `{#if attribute.value.size.or(null)}` and `null` is considered falsy.
